testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2060,6 +2062,58 @@ private String getTableComment(String tableName)
         return null;
     }
 
+    @Test
+    public void testCommentView()
+    {
+        if (!hasBehavior(SUPPORTS_COMMENT_ON_VIEW)) {
+            if (hasBehavior(SUPPORTS_CREATE_VIEW)) {
+                try (TestView view = new TestView(getQueryRunner()::execute, "test_comment_view", "SELECT * FROM region")) {
+                    assertQueryFails("COMMENT ON VIEW " + view.getName() + " IS 'new comment'", "This connector does not support setting view comments");
+                }
+                return;
+            }
+            throw new SkipException("Skipping as connector does not support CREATE VIEW");
+        }
+
```

Comment:
We also need tests which assert failure when COMMENT ON VIEW is used on a table and COMMENT ON TABLE is used on a view.
Ideally COMMENT ON TABLE should no longer work on Hive views and COMMENT ON VIEW should work instead.

Reply:
It's basically covered in `TestCommentTask` now. I can add more connector specific tests when supporting the syntax in those connectors.